### PR TITLE
xcodebuild now autodetects workspace, if present.

### PR DESCRIPTION
**Pull request by dtrenz**
Default behavior of `xcodebuild` --  if no workspace is passed in -- is to use the `.xcodeproj`. This PR enables the `xcodebuild` action to autodetect and use the `.xcworkspace` instead, removing the need to specify the workspace.

### CODE REVIEW

File: lib/fastlane/actions/xcodebuild.rb [hunk 1]
```diff
@@ -157,6 +164,12 @@ def self.hash_to_args(hash)
           end
         end.compact.sort
       end
+
+      def self.detect_workspace
+        # Dir["*.xcworkspace"].first
```
Comment (KrauseFx) [inline lib/fastlane/actions/xcodebuild.rb]: Should be deleted maybe?

File: lib/fastlane/actions/xcodebuild.rb [hunk 2]
```diff
@@ -84,6 +85,12 @@ def self.run(params)
             params[:scheme] ||= scheme
             params[:workspace] ||= workspace
             params[:project] ||= project
+
+            # If no project or workspace was passed in or set as an environment
+            # variable, attempt to autodetect the workspace.
+            if params[:project].to_s.empty? && params[:workspace].to_s.empty?
+              params[:workspace] = detect_workspace()
```
Comment (KrauseFx) [inline lib/fastlane/actions/xcodebuild.rb]: Could you remove the ()

File: lib/fastlane/actions/xcodebuild.rb [hunk 3]
```diff
@@ -157,6 +164,12 @@ def self.hash_to_args(hash)
           end
         end.compact.sort
       end
+
+      def self.detect_workspace
+        # Dir["*.xcworkspace"].first
+        pp Dir.glob("*.xcworkspace")
```
Comment (KrauseFx) [inline lib/fastlane/actions/xcodebuild.rb]: Why the `pp`?

File: lib/fastlane/actions/xcodebuild.rb [hunk 4]
```diff
@@ -157,6 +164,12 @@ def self.hash_to_args(hash)
...
+        Dir.glob("*.xcworkspace").first
```
Comment (KrauseFx) [inline lib/fastlane/actions/xcodebuild.rb]: Shouldn't we show a warning if there were multiple workspaces found?

**Comment by KrauseFx** [comment]
Looks great, thanks, just added a few comments.

File: lib/fastlane/actions/xcodebuild.rb (hunk 1 above)
Comment (dtrenz) [inline lib/fastlane/actions/xcodebuild.rb]: Absolutely.

**Comment by dtrenz** [comment]
I made it so that it will warn when multiple workspaces are present, but it will let you know which one it will use. I didn't think that it should fail if there are multiple workspaces, but I do think that it should be more verbose about which workspace it will use.

What do you think?

**Comment by KrauseFx** [comment]
Perfect, looks great :+1:

**Pull request merged**